pkg/dyninst/gosym/symtab.go

```diff
@@ -89,6 +89,16 @@ func (gst *GoSymbolTable) LocatePC(pc uint64) []GoLocation {
 	return gst.pclntab.locatePC(pc, gst.gofunc)
 }
 
+// FunctionPCResolver builds a FuncPCResolver for the function containing the
+// given program counter (any pc inside the function is valid). Returns nil if
+// the PC could not be resolved to a function.
+//
+// The FuncPCResolver can be used as an alternative to GoSymbolTable.LocatePC()
+// for more efficiently resolving multiple PCs within the same function.
+func (gst *GoSymbolTable) FunctionPCResolver(pc uint64) *FuncPCResolver {
```

Comment:
I think that functions that are fallible but rely on nil checking is a trap. Why not just return errors?